syntax = "proto3";
import "timestamp.proto";

package yaroc;

message Disconnected { string client_name = 1; }

enum EventType {
  Unknown = 0;
  Added = 1;
  Removed = 2;
}

message DeviceEvent {
  string port = 1;
  EventType type = 2;
}

message Coordinates {
  float latitude = 1;
  float longitude = 2;
  float altitude = 3;
  Timestamp time = 4;
}

message MiniCallHome {
  uint32 local_ip = 2;
  float cpu_temperature = 3;
  uint32 freq = 4;
  uint32 min_freq = 5;
  uint32 max_freq = 6;
  float volts = 7;
  int32 signal_dbm = 8;
  int32 signal_snr = 15;
  uint32 cellid = 9;
  uint32 network_type = 10;
  string codes = 11;
  uint64 totaldatarx = 12;
  uint64 totaldatatx = 13;
  Timestamp time = 14;
}

message Status {
  oneof msg {
    Disconnected disconnected = 1;
    MiniCallHome mini_call_home = 2;
    DeviceEvent dev_event = 3;
  }
}
